### PR TITLE
Add plugin watcher design

### PR DESCRIPTION
**Pull request by vikaschoudhary16**
This PR is for adding design of an already merged feature. [Original design was on gdoc](https://docs.google.com/document/d/1dtHpGY-gPe9sY7zzMGnm8Ywo09zJfNH-E1KEALFV39s/edit#heading=h.7fe6spexljh6). With some updates adding the same here for better tracking.

/cc @tengqm @derekwaynecarr @vishh @jiayingz @ConnorDoyle @dchen1107 @RenaudWasTaken  @sjenning

### CODE REVIEW

**Comment by jdumars** [comment]
Is this a KEP?

**Comment by vikaschoudhary16** [comment]
@jdumars This is not in KEP format. Is it required to convert it into KEP?

**Comment by jdumars** [comment]
@vikaschoudhary16 I don't think it's required to be a KEP per se, but I'm not sure why it isn't since the proposal process is being deprecated in favor of KEPs. I'd also like to plug it into the KEP tracking process here: https://github.com/kubernetes-sigs/architecture-tracking/projects/2

**Comment by saad-ali** [comment]
I don't think this needs to be tracked as a KEP since the feature has already been implemented (this is adding the design documentation).

@vikaschoudhary16 is there any difference between what is documented here and what is implemented?

**Comment by vikaschoudhary16** [comment]
@saad-ali There is no difference with implementation.

File: contributors/design-proposals/node/plugin-watcher.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+	// For DevicePlugin, this is the resource name that the plugin manages and
+	// should follow the extended resource name convention.
+	// For CSI, this is the CSI driver registrar name.
+	string name = 2;
+        // Optional endpoint location. If found set by Kubelet component,
+        // Kubelet component will use this endpoint for specific requests.
+        // This allows the plugin to register using one endpoint and possibly use
+        // a different socket for control operations. CSI uses this model to delegate
+        // its registration external from the plugin.
+        string endpoint = 3;
```
Comment (ScorpioCPH) [inline contributors/design-proposals/node/plugin-watcher.md]: nit: these lines are out of alignment.

**Comment by saad-ali** [comment]
Address nits, otherwise:

/lgtm

File: contributors/design-proposals/node/plugin-watcher.md (hunk 1 above)
Comment (vikaschoudhary16) [reply contributors/design-proposals/node/plugin-watcher.md]: @ScorpioCPH not sure, why it is showing dis-alignment here. If you click "view", it shows correct and also if you pull this PR, on local machine also it shows correct. :)

**Comment by saad-ali** [comment]
This has already been implemented in the last quarter.

I'm ok with merge.

/lgtm
/approve

**Pull request merged**